### PR TITLE
build grub-x86_64-efi for i686

### PR DESCRIPTION
**Pull request by sgn**

### CODE REVIEW

**Comment by ericonr** [comment]
Pull in the `help2man` change as well to completely replace #25267 :)

File: srcpkgs/cross-x86_64-linux-musl/template [hunk 1]
```diff
@@ -39,7 +39,9 @@ nostrip_files="libcaf_single.a libgcc.a libgcov.a libgcc_eh.a
  libgnarl_pic.a libgnarl.a libgnat_pic.a libgnat.a libgmem.a"
 depends="${pkgname}-libc-${version}_${revision}"
 
-if [ "$XBPS_TARGET_WORDSIZE" != "64" ]; then
+if [ "${XBPS_TARGET_MACHINE%-musl}" = i686 ]; then
+	:
+elif [ "$XBPS_TARGET_WORDSIZE" != "64" ]; then
 	broken="64-bit crosstoolchain only available on 64-bit host"
 fi
```
Comment (ericonr) [inline srcpkgs/cross-x86_64-linux-musl/template:46]: Is there anything special about `i686` here? We could just remove this limitation.

Comment (sgn) [reply srcpkgs/cross-x86_64-linux-musl/template:46]: I don't know, to be honest :-p

**Comment by sgn** [comment]
I couldn't boot with live usb made from this template, yet. I'll look into this a bit more.

**Comment by sgn** [comment]
My desktop works with i686 Live CD made from this change.

**Comment by fosslinux** [comment]
wait on @q66 's toolchain changes before merging this

**Comment by ericonr** [comment]
Ping

**Comment by fosslinux** [comment]
Are any of the cross changes even needed anymore now that void-cross build-style is being used?

File: srcpkgs/grub/template [hunk 2]
```diff
@@ -63,21 +64,25 @@ do_configure() {
...
 	for _SUPP_BUILD in ${_SUPPLEMENTARY_BUILDS}; do
 		_TARGET=${_SUPP_BUILD%%-*}
 		_PLATFORM=${_SUPP_BUILD##*-}
 		msg_normal "Configuring ${_TARGET} ${_PLATFORM} grub...\n"
 		mkdir $wrksrc/${_PLATFORM}_${_TARGET}_build
 		cd $wrksrc/${_PLATFORM}_${_TARGET}_build
+		if [ "$_TARGET" = x86_64 ] &&
+		   [ "${XBPS_TARGET_MACHINE%-musl}" = i686 ]; then
+			_TARGET=x86_64-linux-musl
+			CFLAGS="${BASE_CFLAGS/-mtune=i686/-mtune=generic}"
```
Comment (q66) [inline srcpkgs/grub/template:82]: i don't like this weird block/substitution, considering the extra platform files are basically arch-independent, there should be no harm in simply configuring them all with empty `CFLAGS`

Comment (q66) [reply srcpkgs/grub/template]: that is not what you did there...

Comment (sgn) [reply srcpkgs/grub/template]: Sorry, but I can't get what you meant!

Comment (q66) [reply srcpkgs/grub/template]: now it looks fine

**Comment by q66** [comment]
hm, what i would do is instead of declaring like a variable `NO_ALTIVEC` or whatever, I would do like:

```
local extra_cflags="-fno-stack-protector"
case "$XBPS_TARGET_MACHINE" in
	ppc*) extra_cflags+=" -mno-altivec";;
esac
....
export CFLAGS+=" ${extra_cflags}"
... configure main platform here ....
...
for supplementary platforms:
    CFLAGS="${extra_cflags}" configure ...
```

which is the same logic as now but imo cleaner

**Pull request merged**